src/framework/languages/internal/languagesservice.cpp

```diff
@@ -63,10 +63,11 @@ void LanguagesService::init()
 
     ValCh<QString> languageCode = configuration()->currentLanguageCode();
     setCurrentLanguage(languageCode.val);
+    m_activeLanguageCode = languageCode.val;
 
-    languageCode.ch.onReceive(this, [this](const QString&) {
+    languageCode.ch.onReceive(this, [this](const QString& newLanguageCode) {
         //! NOTE To change the language at the moment, a restart is required
-        m_needRestartToApplyLanguageChange = true;
+        m_needRestartToApplyLanguageChange = newLanguageCode != m_activeLanguageCode;
```

Comment:
This is needed to make the "Restart required" text go away if the user restores the language explicitly or via **Reset preferences** to the currently active language. Currently, once this text appears, it won't go away.

Reply:
Doesn't the new `m_activeLanguageCode` field duplicate info from `m_currentLanguage.code`?

Reply:
Not exactly. If the language code is "system", `m_currentLanguageCode` will hold the effective language code, e.g. "en-us" so choosing another language and going back to "system" will not make "Restart required" go away if we use `m_currentLanguageCode` here. I am changing `m_activeLanguageCode` to be a local variable in this method since it is not used anywhere else.